cub/cub/agent/agent_scan_by_key.cuh

```diff
@@ -142,11 +142,10 @@ struct AgentScanByKey
 
   using KeyT               = cub::detail::value_t<KeysInputIteratorT>;
   using InputT             = cub::detail::value_t<ValuesInputIteratorT>;
-  using SizeValuePairT     = KeyValuePair<OffsetT, AccumT>;
-  using KeyValuePairT      = KeyValuePair<KeyT, AccumT>;
-  using ReduceBySegmentOpT = ReduceBySegmentOp<ScanOpT>;
+  using FlagValuePairT     = KeyValuePair<int, AccumT>;
+  using ReduceBySegmentOpT = ScanBySegmentOp<ScanOpT>;
 
-  using ScanTileStateT = ReduceByKeyScanTileState<AccumT, OffsetT>;
+  using ScanTileStateT = ReduceByKeyScanTileState<AccumT, int>;
```

Comment:
I had tried making this `bool` but performance dropped for some workloads by `20%`. Using `int` with the logical or, `|`, operator conserved performance and made the algorithms performance almost agnostic to the offset type.